javalib/src/main/scala/java/util/LinkedHashMap.scala

```diff
@@ -0,0 +1,60 @@
+package java.util
+
+import java.{util => ju}
+
+import scala.collection.{mutable => mu}
+
+
+class LinkedHashMap[K, V] private (inner: mu.LinkedHashMap[Box[K], V],
+      initialCapacity: Int, loadFactor: Float, accessOrder: Boolean)
+      extends HashMap[K, V](inner, initialCapacity, loadFactor) {
+
+  self =>
+
+  def this(initialCapacity: Int, loadFactor: Float, accessOrder: Boolean) =
+    this(mu.LinkedHashMap.empty[Box[K], V], initialCapacity, loadFactor, accessOrder)
+
+  def this(initialCapacity: Int, loadFactor: Float) =
+    this(initialCapacity, loadFactor, false)
+
+  def this(initialCapacity: Int) =
+    this(initialCapacity, LinkedHashMap.DEFAULT_LOAD_FACTOR)
+
+  def this() =
+    this(LinkedHashMap.DEFAULT_INITIAL_CAPACITY)
+
+  def this(m: Map[_ <: K, _ <: V]) = {
+    this()
+    initFrom(m)
+  }
+
+  override def get(key: scala.Any): V = {
+    val value = super.get(key)
+    if (accessOrder) {
+      val boxedKey = Box(key.asInstanceOf[K])
+      if (value != null || containsKey(boxedKey)) {
+        inner.remove(boxedKey)
+        inner(boxedKey) = value
```

Comment:
Oh right.

But, hum, then that kind of defeats the purpose of the "optimized" representation for an LRU cache, doesn't it? I mean, the whole idea is that the next call to `get` or `put` is supposed to find the most recently used key. But if we put them at the end, then it will be tried *last*, not first.

This confuses me :-s